api/api_errors.go

```diff
@@ -110,3 +118,44 @@ func (errF3ParticipationTicketStartBeforeExisting) Error() string {
 type errF3NotReady struct{}
 
 func (errF3NotReady) Error() string { return "f3 isn't yet ready to participate" }
+
+type ErrExecutionRevertedWithData struct {
+	Message string
+	Data    string
+}
+
+// Error returns the error message.
+func (e *ErrExecutionRevertedWithData) Error() string { return e.Message }
+
+// FromJSONRPCError converts a JSONRPCError to ErrExecutionRevertedWithData.
+func (e *ErrExecutionRevertedWithData) FromJSONRPCError(jerr jsonrpc.JSONRPCError) error {
+	if jerr.Code != EExecutionRevertedWithData {
+		return nil
+	}
+
+	data, ok := jerr.Data.(string)
```

Comment:
I thought we'd need to unmarshall this from a json binary string? Are we getting a quoted string out of this?